app/models/protocol.rb

```diff
@@ -7,6 +7,7 @@ class Protocol < ApplicationRecord
   validate :at_most_one_stop_measurement
   has_many :protocol_subscriptions, dependent: :destroy
   belongs_to :informed_consent_questionnaire, class_name: 'Questionnaire', optional: true # can be nil
+  belongs_to :language_questionnaire, class_name: 'Questionnaire', optional: true # can be nil
```

Comment:
Instead of making an entire language questionnaire. Would it suffice to simply just make a route where the user can set their locale (as in not using the questionnaire rendering engine, but just a simple page to update one property of their `protocol_subscription`, prefilled with the default setting of their `person`)? What I mean is some setting in a `protocol_subscription` that is a simple boolean that checks, hey did the user already specify a locale that they'd like to use for this protocol, if yes, then fine, if no, redirect them to /language or whatever, which is a site where they can choose between English or Dutch (since those are only two locales supported by the platform), and can be the same for all questionnaires and protocols I think, and then upon saving, do something with that callback that saves the setting for their protocol subscription and then moves on?

Because the proposed approach feels too much like a "double informed consent" to me. If we need to ask anything more than a simple language question -> use an informed questionnaire, and otherwise the page where you set a language can be the same across all protocols, I think.

Reply:
mm, i see what you're saying. I like it.